packages/mco/components/data-policies/create-dr-policy/select-cluster-list.tsx

```diff
@@ -134,18 +164,17 @@ export const SelectClusterList: React.FC<SelectClusterListProps> = ({
           isCompact
           className="mco-select-cluster-list__data-list"
         >
-          {filteredClusters.map((fc) => (
+          {filteredClusters.map((fc, index) => (
             <DataListItem key="">
               <DataListItemRow>
                 <DataListCheck
                   aria-labelledby={t('Checkbox to select cluster')}
+                  id={index.toString()}
                   onChange={onSelect}
-                  value={fc.name}
-                  data-region={fc.region}
-                  isChecked={selectedClusters.some((sc) => sc.name === fc.name)}
+                  isChecked={Object.keys(selectedClusters)?.some((scName) => scName === fc.name)}
                   isDisabled={
-                    selectedClusters.length === MAX_ALLOWED_CLUSTERS &&
-                    !selectedClusters.some((sc) => sc.name === fc.name)
+                    Object.keys(selectedClusters)?.length === MAX_ALLOWED_CLUSTERS &&
+                    !Object.keys(selectedClusters)?.some((scName) => scName === fc.name)
```

Comment:
Make sure "selectedClusters" is  never undefined otherwise it will give errors. 
```suggestion
                    !Object.keys(selectedClusters ?? [])?.some((scName) => scName === fc.name)
```